### PR TITLE
Integrate batching with UDP

### PR DESCRIPTION
**Pull request by renan-**
Hello there,

I'm sending my points over UDP and noticed that points writing to database is very slow because it doesn't use the newly introduced points batcher.
Here's a proposal to integrate it with UDP, please tell me what you think (I'm quite new to Go and InfluxDB so feedback is welcome).

Hoping this will be useful,
Renan

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks @renan- can you sign the CLA?

http://influxdb.com/community/cla.html

**Comment by renan-** [comment]
Done :)

File: udp/udp.go [hunk 1]
```diff
@@ -14,47 +18,98 @@ const (
...
 	return &u
 }
 
-// ListenAndServe binds the server to the given UDP interface.
-func (u *UDPServer) ListenAndServe(iface string) error {
-	addr, err := net.ResolveUDPAddr("udp", iface)
+func (u *UDPServer) SetBatchSize(sz int)             { u.batchSize = sz }
+func (u *UDPServer) SetBatchTimeout(d time.Duration) { u.batchTimeout = d }
+
+// ListenAndServer instructs the UDPServer to start processing data
```
Comment (otoolep) [inline udp/udp.go]: Typo: ListenAndServer --> ListenAndServe

File: udp/udp.go [hunk 2]
```diff
@@ -14,47 +18,98 @@ const (
...
+	u.wg.Add(1)
 	go func() {
+		defer u.wg.Done()
 		for {
-			_, _, err := conn.ReadFromUDP(buf)
+			_, _, err := u.conn.ReadFromUDP(buf)
 			if err != nil {
-				log.Printf("Failed read UDP message: %s.", err)
-				continue
+				close(done)
```
Comment (otoolep) [inline udp/udp.go]: We just added a new `Flush` call to the batcher, which you should add here, just before the call to `close`.

**Comment by otoolep** [comment]
Looks good to me, couple of small changes required.

@corylanou should take a look as well.

**Comment by renan-** [comment]
Updated the code for the new flusher functionality. Cleaned it up as well (it was mainly inspired by the work on Graphite UDP support).

**Comment by otoolep** [comment]
Thanks again @renan- - we are actually undergoing a large refactor of the write path, so will hold off on merging this for a little while.

**Comment by otoolep** [comment]
This no longer applies, since UDP support has been converted to a `Service` and performs batching.

Thanks again @renan-

**Pull request closed without merging**